src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java

```diff
@@ -114,7 +115,7 @@ static Path getIdxPath(Path path) {
 	/** Returns a {@link BAMRecordReader} initialized with the parameters. */
 	@Override public RecordReader<LongWritable,SAMRecordWritable>
 		createRecordReader(InputSplit split, TaskAttemptContext ctx)
-			throws InterruptedException, IOException
+            throws InterruptedException, IOException, ProviderNotFoundException, IllegalArgumentException
```

Comment:
Both ProviderNotFoundException and IllegalArgumentException are runtime exceptions, so they shouldn't be added to the throws clause. Instead, they can be documented in javadoc to indicate the conditions under which they could be thrown by this method.